### PR TITLE
add  user friend list with validation

### PR DESCRIPTION
**Pull request by delowarsikder**
user can access his own friend list. User id show in nav bar

### CODE REVIEW

File: app/controllers/friends_controller.rb [hunk 1]
```diff
@@ -1,12 +1,15 @@
...
+
+  before_action :set_friend, only: [:show, :edit, :update, :destroy]
+  before_action :authenticate_user!, except: [:index, :show]
+  before_action :correct_user, only: [:show, :edit, :update, :destroy]
 
 
   # GET /friends or /friends.json
   def index
-    @friends = Friend.all
+    # @friends = Friend.all
```
Comment (saadrumon) [inline app/controllers/friends_controller.rb:10]: @delowarsikder please remove commented code

File: app/controllers/friends_controller.rb [hunk 2]
```diff
@@ -1,12 +1,15 @@
 class FriendsController < ApplicationController
-  before_action :set_friend, only: %i[ show edit update destroy ]
-  before_action:authenticate_user!,except:[:index, :show]
-  # before_action:correct_user,only:[:edit,:update,:destroy]
+
+  before_action :set_friend, only: [:show, :edit, :update, :destroy]
+  before_action :authenticate_user!, except: [:index, :show]
+  before_action :correct_user, only: [:show, :edit, :update, :destroy]
```
Comment (saadrumon) [inline app/controllers/friends_controller.rb:5]: @delowarsikder This before method is unnecessary

File: app/controllers/friends_controller.rb [hunk 3]
```diff
@@ -63,13 +65,12 @@ def destroy
...
-  #   @friend=current_user.friends.find_by(id: params[:id])
-  #   redirect_to friends_path,notice:"No Authorize acess!" if @friend.nil?
-  # end
-
 
+  ###check the current user ##myself
+  def correct_user
+    @friend = current_user.friends.find_by(id: params[:id])
+    redirect_to friends_path, notice: "Not Authorized To Access This Friend !" if @friend.nil?
+  end
```
Comment (saadrumon) [inline app/controllers/friends_controller.rb:73]: @delowarsikder this method is unnecessary

File: app/views/home/_header.html.erb [hunk 4]
```diff
@@ -32,7 +32,12 @@
         <li class="nav-item">
         <%= link_to "Sign Out", destroy_user_session_path,method: :delete,class:'nav-link'  %>
         </li>        
+        
+        <li class="nav-item">
+        <%= link_to current_user.id, destroy_user_session_path,method: :delete,class:'nav-link'  %>
+        </li>
```
Comment (saadrumon) [inline app/views/home/_header.html.erb:38]: @delowarsikder please remove the redundant sign-out option.

**Review by saadrumon — CHANGES_REQUESTED** [review]

**Comment by delowarsikder** [comment]
This PR is not required anymore.

**Pull request closed without merging**